idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/references/FirReferenceResolveHelper.kt

```diff
@@ -174,10 +171,35 @@ internal object FirReferenceResolveHelper {
             is FirVariableAssignment -> getSymbolsByVariableAssignment(fir, session, symbolBuilder)
             is FirResolvable -> getSymbolsByResolvable(fir, expression, session, symbolBuilder)
             is FirNamedArgumentExpression -> getSymbolsByNameArgumentExpression(expression, analysisSession, symbolBuilder)
+            is FirValueParameter -> getSymbolForVarargParameterType(fir, session, symbolBuilder)
+                ?: handleUnknownFirElement(expression, analysisSession, session, symbolBuilder)
             else -> handleUnknownFirElement(expression, analysisSession, session, symbolBuilder)
         }
     }
 
+    /**
+     * For a vararg parameter, the corresponding FirElement has a fake source kind
+     * [org.jetbrains.kotlin.fir.FirFakeSourceElementKind.ArrayTypeFromVarargParameter]. In this case, [getOrBuildFir] returns the parent
+     * [FirValueParameter] for the parameter type reference KtElement. Therefore, we need this special handling.
+     */
+    private fun getSymbolForVarargParameterType(
+        fir: FirValueParameter,
+        session: FirSession,
+        symbolBuilder: KtSymbolByFirBuilder
+    ): Collection<KtSymbol>? {
+        // Bail out if it's not vararg. Normally, non-vararg parameter should never hit this branch in the first place since there should be
+        // a corresponding `FirResolvedTypeRef` returned for the type reference `KtElement` from `getOrBuildFir`
+        if (!fir.isVararg) return null
+        val elementTypeSymbol = fir
+            .returnTypeRef
+            .coneTypeSafe<ConeKotlinType>()
```

Comment:
Out of curiosity, isn't it safe to just call `fir.returnTypeRef.coneType` ? If it's not safe, that means... some `FirValueParameter`'s return type is unresolved. For checkers side, I did a similar job removing remaining unresolved type ref after FIR resolution. So, I wonder here in IDE, in particular resolve helper, we can assume type ref is resolved (otherwise it's a bug in FIR resolution)?

Reply:
Actually I am not confident if calling `coneTypeSafe` is necessary. I just manually played a bit and looks like calling `coneType` is sufficient.